src/general/PauseMenu.cs

```diff
@@ -0,0 +1,198 @@
+using System;
+using Godot;
+
+/// <summary>
+///   Handles logic in the pause menu
+/// </summary>
+public class PauseMenu : Control
+{
+    [Export]
+    public string HelpCategory;
+
+    [Export]
+    public NodePath PrimaryMenuPath;
+
+    [Export]
+    public NodePath HelpScreenPath;
+
+    [Export]
+    public NodePath LoadMenuPath;
+
+    [Export]
+    public NodePath SaveMenuPath;
+
+    [Export]
+    public NodePath LoadSaveListPath;
+
+    private Control primaryMenu;
+    private HelpScreen helpScreen;
+    private Control loadMenu;
+    private NewSaveMenu saveMenu;
+    private SaveList loadSaveList;
+
+    [Signal]
+    public delegate void OnClosed();
+
+    /// <summary>
+    ///   Triggered when the user hits ESC to open the pause menu
+    /// </summary>
+    [Signal]
+    public delegate void OnOpenWithKeyPress();
+
+    /// <summary>
+    ///   Called when a save needs to be made
+    /// </summary>
+    /// <param name="name">Name of the save to make or empty string</param>
+    [Signal]
+    public delegate void MakeSave(string name);
+
+    public override void _EnterTree()
+    {
+        // This needs to be done early here to make sure the help screen loads the right text
+        helpScreen = GetNode<HelpScreen>(HelpScreenPath);
+        helpScreen.Category = HelpCategory;
+    }
+
+    public override void _Ready()
+    {
+        primaryMenu = GetNode<Control>(PrimaryMenuPath);
+        loadMenu = GetNode<Control>(LoadMenuPath);
+        saveMenu = GetNode<NewSaveMenu>(SaveMenuPath);
+        loadSaveList = GetNode<SaveList>(LoadSaveListPath);
+    }
+
+    public override void _UnhandledInput(InputEvent @event)
+    {
+        if (@event.IsActionPressed("ui_cancel"))
+        {
+            if (Visible)
+            {
+                SetActiveMenu("primary");
+
+                EmitSignal(nameof(OnClosed));
+            }
+            else
+            {
+                EmitSignal(nameof(OnOpenWithKeyPress));
+            }
+        }
+    }
+
+    public void ShowHelpScreen()
+    {
+        SetActiveMenu("help");
```

Comment:
A small thing, the help screen's `RandomizeEasterEgg` method should also be called here.